Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -103,46 +102,40 @@ VerifierOptions verifierOptions
       }
 
       var progressReporter = CreateVerificationProgressReporter(loaded);
+      progressReporter.RecomputeVerificationTree();
       progressReporter.UpdateLastTouchedMethodPositions();
 
       var verificationTasks = await verifier.GetVerificationTasksAsync(loaded, cancellationToken);
       if (VerifierOptions.GutterStatus) {
-        progressReporter.RecomputeVerificationTree();
         progressReporter.ReportRealtimeDiagnostics(false, loaded);
         progressReporter.ReportImplementationsBeforeVerification(
           verificationTasks.Select(t => t.Implementation).ToArray());
       }
 
       var initialViews = new ConcurrentDictionary<ImplementationId, ImplementationView>();
-
       foreach (var task in verificationTasks) {
         var status = StatusFromBoogieStatus(task.CacheStatus);
         var id = GetImplementationId(task.Implementation);
         if (task.CacheStatus is Completed completed) {
           var view = new ImplementationView(task.Implementation.tok.GetLspRange(), status, GetDiagnosticsFromResult(loaded, completed.Result));
           initialViews.TryAdd(GetImplementationId(task.Implementation), view);
         } else if (loaded.ImplementationIdToView.TryGetValue(id, out var existingView)) {
-          initialViews.TryAdd(id, existingView with { Status = status });
+          var view = existingView with { Status = status };
+          initialViews.TryAdd(id, view);
         } else {
           var view = new ImplementationView(task.Implementation.tok.GetLspRange(), status, Array.Empty<Diagnostic>());
           initialViews.TryAdd(GetImplementationId(task.Implementation), view);
         }
       }
 
-      var result = loaded with {
-        CounterexamplesCollector = new ConcurrentStack<Counterexample>(),
-        ImplementationIdToViewCollector = initialViews,
-        VerificationTasks = verificationTasks,
-        ImplementationIdToView = initialViews.ToImmutableDictionary(),
-      };
+      loaded.ImplementationIdToView = initialViews;
+      loaded.VerificationTasks = verificationTasks;
```

Comment:
Is it better to mutate this object in place than to return a new one?  (Is that what the old code did?)

Reply:
The old code had both mutable fields related to verification and sometimes made copies of them, but I think this only gave a false sense of security. In this PR, `DafnyDocument` is designed to be mutated concurrently by verification tasks.